lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -209,18 +202,6 @@ public void finalize() {
 		dispose();
 	}
 
-	private static class HostFailedException extends AblyException {
-		private static final long serialVersionUID = 1L;
-		public HostFailedException(Throwable cause) {
-			super(cause);
-		}		
-		private static AblyException checkFor(Throwable t) {
-			if(t instanceof ConnectException || t instanceof UnknownHostException || t instanceof NoRouteToHostException)
```

Comment:
I preferred this, compared with your proposed change, because here `HostFailedException` knows whether or not a given `Throwable` applies. With your change, that decision is now distant from the declaration of the class.